thread/event.go

```diff
@@ -1,23 +1,39 @@
 package thread
 
 import (
+	"context"
+	"time"
+
+	"github.com/ipfs/go-cid"
 	"github.com/ipfs/go-ipld-format"
+	"github.com/textileio/go-textile-core/crypto"
 )
 
 // Event is the Block format used by Textile threads
 type Event interface {
 	format.Node
 
-	Header() EventHeader
-	Body() format.Node
+	// HeaderID returns the cid of the event header.
+	HeaderID() cid.Cid
+
+	// GetHeader loads and optionally decrypts the event header.
+	// If no key is given, the header time and key methods will return an error.
```

Comment:
So is that saying, keys are mandatory?

Reply:
currently, yeah - some more thought needed around making keys optional